pkg/util/io/io.go

```diff
@@ -20,12 +20,114 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"path/filepath"
+	"time"
 
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/apimachinery/registered"
 	"k8s.io/kubernetes/pkg/runtime"
 )
 
+// ioTimeout is the timeout to use for io operations
+const ioTimeout time.Duration = 5
+
+// Lstat wraps os.Lstat in a timeout.
+func Lstat(path string) (os.FileInfo, error) {
+	return internalLstatTimeout(path, ioTimeout)
+}
+
+func internalLstatTimeout(path string, timeout time.Duration) (os.FileInfo, error) {
+	var (
+		timer      = time.NewTimer(ioTimeout)
+		resChannel = make(chan os.FileInfo)
+		errChannel = make(chan error)
+	)
+
+	defer timer.Stop()
+
+	go func() {
+		fileInfo, err := os.Lstat(path)
+		if err != nil {
+			errChannel <- err
+		} else {
+			resChannel <- fileInfo
+		}
+	}()
+
+	select {
+	case <-timer.C:
+		return nil, fmt.Errorf("Lstat on %v timed out after %v", path, timeout)
```

Comment:
Because NFS.  Create an NFS server, mount something from it, delete/destroy/stop the NFS server, run `ls` on the client. Enjoy the hang...  Its what NFS does...